core/services/keeper/orm.go

```diff
@@ -132,6 +167,16 @@ ORDER BY upkeep_registrations.id ASC, upkeep_registrations.upkeep_id ASC
 	return upkeeps, err
 }
 
+// binaryOfFirstHashInTurn first calculates the first potential head for a turn. It then gets the hash for that head and converts it to binary
+func binaryOfFirstHashInTurn(blockNumber int64, registry Registry, head *types.Head) (string, error) {
+	firstHeadInTurn := blockNumber - (blockNumber % int64(registry.BlockCountPerTurn))
+	hashAtHeight := head.HashAtHeight(firstHeadInTurn)
+	bigInt := new(big.Int)
+	bigInt.SetString(hashAtHeight.Hex(), 0)
+	binaryString := fmt.Sprintf("%b", bigInt)
+	return binaryString, nil
```

Comment:
Can this function ever return an error? We might be missing some error handling

Reply:
no error

Reply:
fixed